GameOfLife.java

```diff
@@ -85,22 +114,50 @@ public static int[][] evolve(int[][] board) {
 	// Assumes that j is at least 1 and at most the number of columns in the board - 1. 
 	// Uses the count(board,i,j) function to count the number of alive neighbors.
 	public static int cellValue(int[][] board, int i, int j) {
-		//// Replace the following statement with your code.
-		return 0;
+		int value = 0;
+		if(board[i][j] == 1){
+			if(count(board, i, j) < 2 || count(board, i, j) > 3){
+				value = 0;
+			}
+			else if(count(board, i, j) == 2 || count(board, i, j) == 3){
+				value = 1;
+			}
+		}
+		else if(board[i][j] == 0){
+			if(count(board, i, j) == 3){
+				value = 1;
+			}
+		}
+		return value;
 	}
 	
 	// Counts and returns the number of living neighbors of the given cell
 	// (The cell itself is not counted).
 	// Assumes that i is at least 1 and at most the number of rows in the board - 1. 
 	// Assumes that j is at least 1 and at most the number of columns in the board - 1. 
 	public static int count(int[][] board, int i, int j) {
-		//// Replace the following statement with your code.
-		return 0;
+		int count = 0;
+		for(int row = i - 1; row <= i + 1; row++){
+			for(int cols = j - 1; cols <= j + 1; cols++){
+					if(board[row][cols] == 1){
+						count++;
+						if(row == i && cols == j && board[i][j] == 1){
+							count--;
+					}
+				}
```

Comment:
Better to not count it at all instead of count this case than discount it.
```suggestion
					if((row != i || cols != j) && board[row][cols] == 1){
						count++;
					}
				}
```